### PR TITLE
samples:simple_embedding: Enable bazel dylib samples

### PR DESCRIPTION
**Pull request by hcindyl**

### CODE REVIEW

File: iree/samples/simple_embedding/BUILD [hunk 1]
```diff
@@ -70,112 +70,110 @@ if(${IREE_HAL_DRIVER_DYLIB} AND (${IREE_TARGET_BACKEND_DYLIB-LLVM-AOT} OR DEFINE
...
+        "//iree/vm",
+        "//iree/vm:bytecode_module",
+    ],
+)
+
+iree_bytecode_module(
+    name = "simple_embedding_test_bytecode_module_dylib_x86_64",
+    src = "simple_embedding_test.mlir",
+    c_identifier = "iree_samples_simple_embedding_test_module_dylib_x86_64",
+    flags = [
```
Comment (benvanik) [inline iree/samples/simple_embedding/BUILD:100]: here's a few more flags (I'll make a rollup at some point so we can just say `-iree-strip-all` or something):

```
-iree-vm-bytecode-module-strip-symbols=true
-iree-vm-bytecode-module-strip-source-map=true
-iree-vm-emit-polyglot-zip=false
```
You lose nice stack traces and such, but that seems fine in this simple_embedding demo and give you smaller binaries!

**Review by benvanik — CHANGES_REQUESTED** [review]

File: iree/samples/simple_embedding/BUILD (hunk 1 above)
Comment (hcindyl) [reply iree/samples/simple_embedding/BUILD:100]: Adding `-iree-vm-bytecode-module-strip-symblos=true` causes execution failure

(at rv32-baremetal)
```
/usr/local/google/home/hcindyl/Projects/iree-dev/iree/iree/vm/bytecode_module.c:166: INVALID_ARGUMENT; mismatched internal_functions/function_descriptors vectors (zu != zu)
```

at x86_64
```
$ ./simple_embedding_dylib 
/usr/local/google/home/hcindyl/Projects/iree-dev/iree/iree/vm/bytecode_module.c:166: INVALID_ARGUMENT; mismatched internal_functions/function_descriptors vectors (0 != 2)
simple_embedding done
```

I will file an issue.

Comment (benvanik) [reply iree/samples/simple_embedding/BUILD:100]: Cool! Can fix that, and then this using it will test it :P

Comment (hcindyl) [reply iree/samples/simple_embedding/BUILD:101]: Flags are added (the problematic one is commented out)

**Review by benvanik — APPROVED** [review]

**Pull request merged**